src/backend/lima.ts

```diff
@@ -2154,8 +2191,35 @@ class LimaKubernetesBackend extends events.EventEmitter implements K8s.Kubernete
     await this.vm.writeFile('/etc/logrotate.d/k3s', LOGROTATE_K3S_SCRIPT);
   }
 
-  protected async deleteIncompatibleData(isDowngrade: boolean) {
-    if (isDowngrade) {
+  /**
+   * Persist the given version into the VM disk, so we can look it up later.
+   */
+  protected async persistVersion(version: semver.SemVer): Promise<void> {
+    const filepath = '/var/lib/rancher/k3s/version';
+
+    await this.vm.writeFile(filepath, version.version);
+  }
+
+  /**
+   * Look up the previously persisted version.
+   */
+  protected async getPersistedVersion(): Promise<ShortVersion | undefined> {
+    const filepath = '/var/lib/rancher/k3s/version';
+
+    try {
+      return await this.vm.readFile(filepath);
+    } catch (ex) {
+      return undefined;
+    }
+  }
+
+  protected async deleteIncompatibleData(desiredVersion: semver.SemVer) {
+    const existingVersion = await this.getPersistedVersion();
+
+    if (!existingVersion) {
+      return;
+    }
```

Comment:
Doesn't this mean that when you downgrade k3s after updating RD, you will not delete the incompatible data, even though you should?

Reply:
Switched to using `k3s --version` instead, so this is no longer relevant.